tools/flaky-test-reporter/json-report.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+
+	"github.com/knative/test-infra/shared/prow"
+)
+
+type JSONReport struct {
+	Flaky []string `json:"flaky"`
+}
+
+func (jr *JSONReport) writeToArtifactsDir(repo string, dryrun bool) error {
+	artifactsDir := prow.GetLocalArtifactsDir()
+	// no need to create the directory, it is guaranteed to exist since artifacts were generated earlier
+	outFilePath := path.Join(artifactsDir, repo, "all-flaky-tests.json")
```

Comment:
is there a non-all option?
```suggestion
	outFilePath := path.Join(artifactsDir, repo, "flaky-tests.json")
```